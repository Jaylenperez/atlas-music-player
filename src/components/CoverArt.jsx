import cover from "../assets/placeholder.svg";

export function CoverArt() {
  return (
    <div className="pb-5">
      <img src={cover} alt="Cover Art" className="size-100" />
    </div>
  );
}
