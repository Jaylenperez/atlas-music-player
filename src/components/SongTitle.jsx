import React from "react";

export function SongTitle() {
  return (
    <div className="w-full">
      <h2 className="text-left text-2xl font-bold text-sky-300">Song title</h2>
      <p className="text-left text-lg text-stone-400">Artist</p>
    </div>
  );
}
