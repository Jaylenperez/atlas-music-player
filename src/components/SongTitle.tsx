// src/components/SongTitle.tsx
import React from "react";

export interface SongTitleProps {
  lightMode: boolean;
  title: string;
  artist: string;
}

export const SongTitle: React.FC<SongTitleProps> = ({
  lightMode,
  title,
  artist,
}) => (
  <div className="w-full">
    <h2
      className={`text-left text-2xl font-bold ${
        lightMode ? "text-slate-950" : "text-sky-400"
      }`}
    >
      {title}
    </h2>
    <p
      className={`text-left text-lg ${
        lightMode ? "text-yellow-950" : "text-fuchsia-500"
      }`}
    >
      {artist}
    </p>
  </div>
);
