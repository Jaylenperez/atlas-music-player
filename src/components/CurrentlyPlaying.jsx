// src/components/CurrentlyPlaying.jsx
import { CoverArt } from "./CoverArt";
import { SongTitle } from "./SongTitle";
import { PlayControls } from "./PlayControls";
import { VolumeControls } from "./VolumeControls";

export default function CurrentlyPlaying({ lightMode }) {
  return (
    <div className="flex h-full w-full flex-col justify-between px-4 py-4">
      {/* Top: cover art */}
      <div className="flex-shrink-0">
        <CoverArt />
      </div>

      {/* Middle: song title + play/pause/etc */}
      <div className="space-y-3">
        <SongTitle lightMode={lightMode} />
        <PlayControls />
      </div>

      {/* Bottom: volume slider */}
      <div className="flex-shrink-0">
        <VolumeControls />
      </div>
    </div>
  );
}
