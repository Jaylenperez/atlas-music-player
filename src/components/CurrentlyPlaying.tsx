// src/components/CurrentlyPlaying.tsx
import React, { useEffect, useState } from "react";
import { CoverArt } from "./CoverArt";
import { SongTitle } from "./SongTitle";
import { PlayControls } from "./PlayControls";
import { VolumeControls } from "./VolumeControls";

interface PlaylistSong {
  id: string;
  title: string;
  artist: string;
  genre: string;
  duration: number;
}

interface Song extends PlaylistSong {
  cover: string;
  song: string;
}

interface CurrentlyPlayingProps {
  lightMode: boolean;
  track: PlaylistSong;
}

export const CurrentlyPlaying: React.FC<CurrentlyPlayingProps> = ({
  lightMode,
  track,
}) => {
  // we'll store the full Song here once we fetch it
  const [fullSong, setFullSong] = useState<Song | null>(null);

  useEffect(() => {
    // every time `track.id` changes, fetch its details
    fetch(`/api/v1/songs/${track.id}`)
      .then((res) => {
        if (!res.ok) throw new Error("Failed to load song details");
        return res.json();
      })
      .then((data: Song) => setFullSong(data))
      .catch((err) => {
        console.error(err);
        setFullSong(null);
      });
  }, [track.id]);

  // while loading, you can show a spinner or fallback image
  const coverSrc = fullSong?.cover ?? "/placeholder.svg";

  return (
    <div className="flex h-full w-full flex-col justify-between px-4 py-4">
      <div className="flex-shrink-0">
        <CoverArt src={coverSrc} />
      </div>

      {fullSong && (
        <div className="space-y-3">
          <SongTitle
            lightMode={lightMode}
            title={fullSong.title}
            artist={fullSong.artist}
          />
          <PlayControls lightMode={lightMode} />
        </div>
      )}

      <div className="flex-shrink-0">
        <VolumeControls lightMode={lightMode} />
      </div>
    </div>
  );
};
