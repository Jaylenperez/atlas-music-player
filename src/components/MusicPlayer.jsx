import { CoverArt } from "./CoverArt";
import { SongTtile } from "./SongTitle";
import { PlayControls } from "./PlayControls";
import { VolumeControls } from "./VolumeControls";
import PlayListItem from "./PlayListItem";

export default function MusicPlayer() {
  return (
    <div>
      <CoverArt />
      <SongTtile />
      <PlayControls />
      <VolumeControls />
      <PlayListItem
        title="Nightmare"
        artist="Avenged Sevenfold"
        length="6:14"
      />
    </div>
  );
}
