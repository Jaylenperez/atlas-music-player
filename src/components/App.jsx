// src/App.jsx
import React, { useState } from "react";
import CurrentlyPlaying from "./CurrentlyPlaying";
import Playlist from "./Playlist";

export default function App() {
  const [lightMode, setLightMode] = useState(false);

  return (
    <div
      className={`${
        lightMode ? "bg-gray-100 text-gray-900" : "bg-indigo-950 text-white"
      } flex h-screen flex-col overflow-x-hidden md:flex-row`}
    >
      {/* toggle button */}
      <button
        onClick={() => setLightMode(!lightMode)}
        className={`absolute top-4 right-4 rounded px-3 py-1 text-sm font-medium ${
          lightMode ? "bg-slate-400 text-gray-800" : "bg-gray-800 text-gray-200"
        } transition`}
      >
        {lightMode ? "Dark Mode" : "Light Mode"}
      </button>

      {/* left panel */}
      <div className="flex w-full flex-col justify-between p-4 md:w-1/2 md:p-8">
        <CurrentlyPlaying />
      </div>

      {/* right panel */}
      <div className="w-full overflow-y-auto border-t border-gray-200 p-4 md:w-1/2 md:border-t-0 md:border-l md:p-8">
        <Playlist lightMode={lightMode} />
      </div>
    </div>
  );
}
