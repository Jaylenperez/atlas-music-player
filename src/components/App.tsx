import React, { useState, useEffect } from "react";
import { CurrentlyPlaying } from "./CurrentlyPlaying";
import Playlist from "./Playlist";

export default function App() {
  const [lightMode, setLightMode] = useState(false);

  // ← new state for the full playlist and the selected item
  const [playlist, setPlaylist] = useState([]);
  const [selectedTrack, setSelectedTrack] = useState(null);

  useEffect(() => {
    fetch("/api/v1/playlist")
      .then((res) => res.json())
      .then((data) => {
        setPlaylist(data);
        if (data.length > 0) setSelectedTrack(data[0]);
      })
      .catch((err) => console.error("Failed to load playlist:", err));
  }, []);

  return (
    <div
      className={`${
        lightMode ? "bg-red-50 text-gray-900" : "bg-indigo-950 text-white"
      } flex h-screen flex-col overflow-x-hidden md:flex-row`}
    >
      <button
        onClick={() => setLightMode(!lightMode)}
        className={`absolute top-4 right-4 rounded px-3 py-1 text-sm font-medium ${
          lightMode ? "bg-slate-400 text-gray-800" : "bg-gray-800 text-gray-200"
        } transition`}
      >
        {lightMode ? "Dark Mode" : "Light Mode"}
      </button>

      <div className="flex w-full flex-col justify-between p-4 md:w-1/2 md:p-8">
        {/* pass the entire track object (including cover, duration, etc.) */}
        {selectedTrack && (
          <CurrentlyPlaying lightMode={lightMode} track={selectedTrack} />
        )}
      </div>

      <div className="w-full overflow-y-auto border-t border-gray-200 p-4 md:w-1/2 md:border-t-0 md:border-l md:p-8">
        <Playlist
          lightMode={lightMode}
          tracks={playlist}
          selected={selectedTrack}
          onSelect={setSelectedTrack}
        />
      </div>
    </div>
  );
}
