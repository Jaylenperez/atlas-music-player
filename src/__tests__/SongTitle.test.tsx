// src/__tests__/SongTitle.test.tsx
import React from "react";
import { render, screen } from "@testing-library/react";
import { describe, test, expect } from "vitest";
import { SongTitle } from "../components/SongTitle";

describe("SongTitle component", () => {
  test("matches snapshot in light mode", () => {
    const { container } = render(
      <SongTitle lightMode={true} title="Dreamscape" artist="Aurora" />
    );
    expect(container).toMatchSnapshot();
  });

  test("matches snapshot in dark mode", () => {
    const { container } = render(
      <SongTitle lightMode={false} title="Dreamscape" artist="Aurora" />
    );
    expect(container).toMatchSnapshot();
  });

  test("renders title and artist text correctly", () => {
    render(
      <SongTitle lightMode={true} title="Windsong" artist="Skyline" />
    );
    expect(screen.getByText("Windsong")).toBeInTheDocument();
    expect(screen.getByText("Skyline")).toBeInTheDocument();
  });

  test("applies the correct classes based on lightMode", () => {
    // Light mode
    const { rerender } = render(
      <SongTitle lightMode={true} title="Echoes" artist="Reverie" />
    );
    const titleEl = screen.getByRole("heading", { level: 2 });
    const artistEl = screen.getByText("Reverie");

    expect(titleEl).toHaveClass("text-slate-950");
    expect(artistEl).toHaveClass("text-yellow-950");

    // Dark mode
    rerender(
      <SongTitle lightMode={false} title="Echoes" artist="Reverie" />
    );
    expect(titleEl).toHaveClass("text-sky-400");
    expect(artistEl).toHaveClass("text-fuchsia-500");
  });
});
