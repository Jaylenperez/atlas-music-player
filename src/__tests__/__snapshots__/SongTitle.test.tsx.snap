// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle component > matches snapshot in dark mode 1`] = `
<div>
  <div
    class="w-full"
  >
    <h2
      class="text-left text-2xl font-bold text-sky-400"
    >
      Dreamscape
    </h2>
    <p
      class="text-left text-lg text-fuchsia-500"
    >
      Aurora
    </p>
  </div>
</div>
`;

exports[`SongTitle component > matches snapshot in light mode 1`] = `
<div>
  <div
    class="w-full"
  >
    <h2
      class="text-left text-2xl font-bold text-slate-950"
    >
      Dreamscape
    </h2>
    <p
      class="text-left text-lg text-yellow-950"
    >
      Aurora
    </p>
  </div>
</div>
`;
